components/layout_2020/fragment_tree/box_fragment.rs

```diff
@@ -80,6 +86,10 @@ pub(crate) struct BoxFragment {
 
     #[serde(skip_serializing)]
     pub background_mode: BackgroundMode,
+
+    /// Additional information of from layout that could be used by Javascripts and devtools.
+    #[serde(skip_serializing)]
+    pub detailed_layout_info: Option<DetailedLayoutInfo>,
```

Comment:
Naming nit: `SpecificLayoutInfo` might be a better description. I like the way you've designed this as we can start putting information for other layout modes here as well.